src/main/java/org/peergos/Client.java

```diff
@@ -79,7 +79,7 @@ public Client(Args args) throws Exception {
         InetSocketAddress localAPIAddress = new InetSocketAddress(apiAddress.getHost(), apiAddress.getPort());
 
         int maxConnectionQueue = 500;
-        int handlerThreads = 50;
+        int handlerThreads = 5;
```

Comment:
Ha that explains it. This is fine but optional - ideally it could be configured via environment variable or script arg.